sympy/concrete/summations.py

```diff
@@ -985,7 +986,7 @@ def eval_sum(f, limits):
     from sympy.functions import KroneckerDelta
 
     (i, a, b) = limits
-    if f.is_zero:
+    if f.is_zero or f.simplify().is_zero:
```

Comment:
It seems as if this line would be the reason f

Reply:
`is_zero` can return `True` if the value is known to be zero, `False` if it is known to not be zero or `None` if it is not known if it is zero or not.

`None` is actually quite common. For example a plain symbol will return `None`, so most symbolic expressions will return it. `simplify` will in some cases make expressions that earlier returned `None`, return `True`, but quite rarely for symbolic expressions.

Reply:
It seems like you have triggered a bug elsewhere though. Would be interesting to know which expression it is that cannot be called with `simplify()`, because that is what seems to be the problem.

Reply:
yess exactly that's what is happening , cause even if we look at the truth table or so as it is called , if f.is_zero is True then the statement will execute , if f.is_zero is False then I dont' think f.simplify().is_zero would ever give True. If f.is_zero is None then f.simplify().is_zero could give True which is the case we want to fix eg Sum(sin(x)**2 +cos(x)**2 - 1 ,(x, 1, oo)) etc etc 
It seems like the following code block from core/test_diff.py is the only reason due to which both the Travis test's are failing
  ```
  exprm = x*sin(x)
    mul_diff = diff(exprm, (x, n))
    assert isinstance(mul_diff, Sum)
    for i in range(5):
        assert mul_diff.subs(n, i).doit() == exprm.diff((x, i)).expand()

    exprm2 = 2*y*x*sin(x)*cos(x)*log(x)*exp(x)
    dex = exprm2.diff((x, n))
    assert isinstance(dex, Sum)
    for i in range(7):
        assert dex.subs(n, i).doit().expand() == \
        exprm2.diff((x, i)).expand()
```

Reply:
> `f.is_zero `gives `None` and `f.simplify().is_zero` gives `True`

That is expected behaviour as the `is_zero` check should be "quick" and `simplify` can be more time-consuming. See, e.g., the discussion in #21979. So not really a bug as such.

I do not get an error when I test with similar changes locally, so hard to tell exactly the expression (the code is one thing, the exact expression of `f` where `f.simplify()` is failing is the most interesting one).

Reply:
Gah! I guess one actually have to change the code, not just add a `print`...

Turns out that `x*Derivative(sin(x), (x, -_d))/(factorial(-_d)*factorial(_d))` cannot be simplified.

Reply:
I opened #22200 for this.

Reply:
> Gah! I guess one actually have to change the code, not just add a `print`...
> 
> Turns out that `x*Derivative(sin(x), (x, -_d))/(factorial(-_d)*factorial(_d))` cannot be simplified.

ahh ok ok now I get it , I tried multiple things to check what's break but wasn't getting any success .

Reply:
> I opened #22200 for this.

wow, thanks for this . Now that the you've reported the appropriate error and original issue which was to fix the `is_convergent()` method for some trigo and related expressions has been addressed ...... I'll make relevant changes as requested by other reviewers and check for proper tests regarding the main issue . We could probably implement the `doit() `method for such trigo expressions after that issue has been fixed ( I'll leave a comment on that issue as a reminder) as that wasn't the main point of this issue (was just something I discovered along the way ) .This may be then merged after a review again !